app/controllers/delete_controller.rb

```diff
@@ -13,7 +13,9 @@ def destroy
     end
 
     email = current_user.email
+    event = record_security_event(SecurityActivity::USER_DESTROYED, user: current_user)
     current_user.destroy!
+    event.destroy! if current_user.persisted?
```

Comment:
I'm not really sure about this, if the destroy were to fail for whatever reason an event would have _already_ been logged to splunk.

At least we can clear it up from the local DB, though perhaps that's a smell worth keeping around 🤔 .
Ideally i'd like to run the `record_security_event` after the current user has been destroyed, but the validations complains due to the validations...

Reply:
What do you think?

Reply:
The `current_user.destroy!` will take care of destroying the events:

https://github.com/alphagov/govuk-account-manager-prototype/blob/5e85bb865caccab4d33a8b12bf4a8b3cd3e57ef4/app/models/user.rb#L28-L29

Reply:
Ah yes, but if the `current_user.destroy!` failed or was refused, then wouldn't this event hang around?
So we'd have an event saying it was destroyed when it wasn't.

It's  abit moot as we would have sent that to splunk by then...

Reply:
I guess we could have an option with `record_event` to just create the event and not `log` it, then do this:

```
User.transaction do
  event_json = record_security_event(SecurityActivity::USER_DESTROYED, user: current_user, log_to_splunk: false).to_hash.to_json
  current_user.destroy!
end

Rails.logger.public_send Rails.application.config.log_level, event_json
```

So it only get logged after the transaction completes.

Reply:
I also tried an option of using an after destroy active record callback but then we lose the requset data.

That strategy above looks great though, will give that a go!